kraken-std/src/kraken/std/buffrs/tasks.py

```diff
@@ -0,0 +1,136 @@
+from __future__ import annotations
+
+import contextlib
+import enum
+import logging
+import os
+import subprocess as sp
+
+from kraken.common import CredentialsWithHost, atomic_file_swap
+from kraken.core import BackgroundTask, Project, Property, Task, TaskStatus
+
+from .manifest import BuffrsManifest
+
+logger = logging.getLogger(__name__)
+
+
+class Language(enum.Enum):
+    PYTHON = "python"
+
+
+class BuffrsLoginTask(Task):
+    """This task logs into artifactory with buffrs"""
+
+    description = "Login to artifactory with buffrs."
+    artifactory_credentials: Property[CredentialsWithHost]
+
+    def execute(self) -> TaskStatus:
+        credentials = self.artifactory_credentials.get()
+
+        command = ["buffrs", "login", "--registry", credentials.host]
+
+        return TaskStatus.from_exit_code(
+            command,
+            sp.run(
+                command,
+                cwd=self.project.directory,
+                env=os.environ.copy(),
+                input=credentials.password,
+                text=True,
+            ).returncode,
+        )
+
+
+class BuffrsInstallTask(Task):
+    """Install dependencies defined in `Proto.toml`"""
+
+    description = "Runs `buffrs install` to download protobuf dependencies"
+
+    def execute(self) -> TaskStatus:
+        command = ["buffrs", "install"]
+
+        return TaskStatus.from_exit_code(
+            command,
+            sp.call(
+                command,
+                cwd=self.project.directory,
+                env=os.environ.copy(),
+            ),
+        )
+
+
+class BuffrsBumpVersionTask(BackgroundTask):
```

Comment:
@alexespencer I vaguely remember a thread on buffrs about not having the version number in the Buffrs manifest, I just want to confirm that buffrs still does store the version there?

If `buffrs publish` would accept a `--version` argument, this temporary version bump would not be necessary.

Reply:
If we do still need the version bump in the file, I would suggest we just make that logic part of the publish task. I don't think there's any other task that requires the version to be bumped?

Reply:
Hey Niklas, publish does not support `--version` - so I'll move this login into the publish task

Reply:
Just to clarify, the login is probably good to have as a separate task. People can run `krakenw run buffrs.login` to explicitly propagate their Buffrs credentials. :) This is about the version bump.